### PR TITLE
Code review 2

### PR DESCRIPTION
**Pull request by vishalrishu**
- Shifted some db queries from controller to model.
- Added some validation to handle errors like if there isn't a post with id passed in params.

### CODE REVIEW

File: app/views/posts/index.js.erb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+$('#posts').append('<%= escape_javascript render(@posts) %>');
+$('.pagination').replaceWith('<%= escape_javascript will_paginate(@posts) %>');
```
Comment (abhishek-8) [inline app/views/posts/index.js.erb:2]: Try to not follow this for further tasks. Do not use ruby variables directly in js. Looks for ways to do on callbacks of ajax requests.

Comment (vishalrishu) [reply app/views/posts/index.js.erb:2]: Ok. Will check that

File: app/models/user.rb [hunk 2]
```diff
@@ -5,4 +5,10 @@ class User < ApplicationRecord
        :recoverable, :rememberable, :trackable, :validatable
   has_many :posts, dependent: :destroy
   has_one_attached :avatar
+
+  def self.get_posts(user_id)
```
Comment (abhishek-8) [inline app/models/user.rb:9]: This should be an instance method

Comment (vishalrishu) [reply app/models/user.rb]: Done

**Pull request merged**